modules/fx/arrow-fx/src/test/kotlin/arrow/fx/IOTest.kt

```diff
@@ -460,14 +475,20 @@ class IOTest : UnitSpec() {
     "unsafeRunAsyncCancellable can cancel even for infinite asyncs" {
```

Comment:
This changes the behavior of the test.
Now it's calling the `cb` in `guaranteeCase`, and before it was being called by `unsafeRunAsyncCancellable` when `OnCancel.ThrowCancellationException` occurs (by default this is `OnCancel.Silent`). Although it looks like it was not working because the new assertion is correct `shouldBe 1.some()`.

I need to double check this again, I remember looking at this a couple times. It should probably be rewritten to be a little simpler.